spec/rubocop/cop/layout/class_structure_spec.rb

```diff
@@ -321,4 +318,27 @@ def do_something
       end
     RUBY
   end
+
+  it 'registers an offense and corrects when private method with inline modifier is declared before public' do
+    expect_offense(<<~RUBY)
+      class A
+        private def foo
+        end
+
+        public def bar
```

Comment:
Does this still flag and autocorrect the offense if the explicit `public` isn't there? Is that captured in a test?

Reply:
Thanks, I've added the test